### DISCUSSION TITLE
prometheus - "error starting web server: accept tcp 0.0.0.0:9090: accept: invalid argument"

### DISCUSSION THREAD

**Discussion by alexrlima81** [prometheus/prometheus#10482] [Q&A]
error to start prometheus / error starting web server
I ran the promtool and checked OK, but failed to start prometheus with ./prometheus 
./promtool check config prometheus.yml
Checking prometheus.yml
 SUCCESS: prometheus.yml is valid prometheus config file syntax

./prometheus --config.file=prometheus.yml
ts=2022-03-23T14:16:35.054Z caller=main.go:479 level=info msg="No time or size retention was set so using the default time retention" duration=15d
ts=2022-03-23T14:16:35.054Z caller=main.go:516 level=info msg="Starting Prometheus" version="(version=2.34.0, branch=HEAD, revision=881111fec4332c33094a6fb2680c71fffc427275              )"
ts=2022-03-23T14:16:35.054Z caller=main.go:518 level=warn msg="This Prometheus binary has not been compiled for a 64-bit architecture. Due to virtual memory constraints of               32-bit systems, it is highly recommended to switch to a 64-bit binary of Prometheus." GOARCH=386
ts=2022-03-23T14:16:35.054Z caller=main.go:521 level=info build_context="(go=go1.17.8, user=root@121ad7ea5487, date=20220315-15:11:06)"
ts=2022-03-23T14:16:35.054Z caller=main.go:522 level=info host_details="(Linux 2.6.18-308.1.1.el5PAE #1 SMP Fri Feb 17 16:55:17 EST 2012 i686  (none))"
ts=2022-03-23T14:16:35.054Z caller=main.go:523 level=info fd_limits="(soft=1024, hard=1024)"
ts=2022-03-23T14:16:35.054Z caller=main.go:524 level=info vm_limits="(soft=2147483647b, hard=2147483647b)"
ts=2022-03-23T14:16:35.057Z caller=web.go:540 level=info component=web msg="Start listening for connections" address=0.0.0.0:9090
ts=2022-03-23T14:16:35.064Z caller=main.go:937 level=info msg="Starting TSDB ..."
ts=2022-03-23T14:16:35.064Z caller=dir_locker.go:77 level=warn component=tsdb msg="A lockfile from a previous execution already existed. It was replaced" file=/opt/promethe              us-2.34.0.linux-386/data/lock
ts=2022-03-23T14:16:35.065Z caller=tls_config.go:195 level=info component=web msg="TLS is disabled." http2=false
ts=2022-03-23T14:16:35.065Z caller=main.go:799 level=info msg="Stopping scrape discovery manager..."
ts=2022-03-23T14:16:35.066Z caller=main.go:813 level=info msg="Stopping notify discovery manager..."
ts=2022-03-23T14:16:35.066Z caller=main.go:835 level=info msg="Stopping scrape manager..."
ts=2022-03-23T14:16:35.066Z caller=main.go:809 level=info msg="Notify discovery manager stopped"
ts=2022-03-23T14:16:35.066Z caller=main.go:795 level=info msg="Scrape discovery manager stopped"
ts=2022-03-23T14:16:35.066Z caller=main.go:829 level=info msg="Scrape manager stopped"
ts=2022-03-23T14:16:35.066Z caller=manager.go:946 level=info component="rule manager" msg="Stopping rule manager..."
ts=2022-03-23T14:16:35.066Z caller=manager.go:956 level=info component="rule manager" msg="Rule manager stopped"
ts=2022-03-23T14:16:35.066Z caller=notifier.go:600 level=info component=notifier msg="Stopping notification manager..."
ts=2022-03-23T14:16:35.066Z caller=main.go:1068 level=info msg="Notifier manager stopped"
ts=2022-03-23T14:16:35.067Z caller=head.go:493 level=info component=tsdb msg="Replaying on-disk memory mappable chunks if any"
ts=2022-03-23T14:16:35.067Z caller=head.go:536 level=info component=tsdb msg="On-disk memory mappable chunks replay completed" duration=5.208µs
ts=2022-03-23T14:16:35.067Z caller=head.go:542 level=info component=tsdb msg="Replaying WAL, this may take a while"
ts=2022-03-23T14:16:35.068Z caller=head.go:613 level=info component=tsdb msg="WAL segment loaded" segment=0 maxSegment=8
ts=2022-03-23T14:16:35.068Z caller=head.go:613 level=info component=tsdb msg="WAL segment loaded" segment=1 maxSegment=8
ts=2022-03-23T14:16:35.069Z caller=head.go:613 level=info component=tsdb msg="WAL segment loaded" segment=2 maxSegment=8
ts=2022-03-23T14:16:35.070Z caller=head.go:613 level=info component=tsdb msg="WAL segment loaded" segment=3 maxSegment=8
ts=2022-03-23T14:16:35.070Z caller=head.go:613 level=info component=tsdb msg="WAL segment loaded" segment=4 maxSegment=8
ts=2022-03-23T14:16:35.071Z caller=head.go:613 level=info component=tsdb msg="WAL segment loaded" segment=5 maxSegment=8
ts=2022-03-23T14:16:35.071Z caller=head.go:613 level=info component=tsdb msg="WAL segment loaded" segment=6 maxSegment=8
ts=2022-03-23T14:16:35.071Z caller=head.go:613 level=info component=tsdb msg="WAL segment loaded" segment=7 maxSegment=8
ts=2022-03-23T14:16:35.072Z caller=head.go:613 level=info component=tsdb msg="WAL segment loaded" segment=8 maxSegment=8
ts=2022-03-23T14:16:35.072Z caller=head.go:619 level=info component=tsdb msg="WAL replay completed" checkpoint_replay_duration=51.411µs wal_replay_duration=4.189993ms total              _replay_duration=4.269721ms
ts=2022-03-23T14:16:35.073Z caller=main.go:958 level=info fs_type=EXT4_SUPER_MAGIC
ts=2022-03-23T14:16:35.073Z caller=main.go:961 level=info msg="TSDB started"
ts=2022-03-23T14:16:35.073Z caller=main.go:1077 level=error err="error starting web server: accept tcp 0.0.0.0:9090: accept: invalid argument"

**Comment by roidelapluie** [comment]
Why are you using the 32-bit edition of Prometheus? it would be better to switch to amd64.


is it possible that you do not have IPv4 built-in? Could you try: `prometheus --web.listen-address '[::1]:9090'`

**Reply by alexrlima81** [reply]
I have Prometheus working on some 64bit hosts, but I have some old hosts on 32bit. I will try to make the upgrade.

I ran the command ./prometheus --web.listen-address "[::1]:9090"

./prometheus --web.listen-address "[::1]:9090"
ts=2022-03-24T13:47:33.260Z caller=main.go:479 level=info msg="No time or size retention was set so using the default time retention" duration=15d
ts=2022-03-24T13:47:33.260Z caller=main.go:516 level=info msg="Starting Prometheus" version="(version=2.34.0, branch=HEAD, revision=881111fec4332c33094a6fb2680c71fffc427275)"
ts=2022-03-24T13:47:33.260Z caller=main.go:518 level=warn msg="This Prometheus binary has not been compiled for a 64-bit architecture. Due to virtual memory constraints of 32-bit systems, it is highly recommended to switch to a 64-bit binary of Prometheus." GOARCH=386
ts=2022-03-24T13:47:33.260Z caller=main.go:521 level=info build_context="(go=go1.17.8, user=root@121ad7ea5487, date=20220315-15:11:06)"
ts=2022-03-24T13:47:33.260Z caller=main.go:522 level=info host_details="(Linux 2.6.18-308.1.1.el5PAE #1 SMP Fri Feb 17 16:55:17 EST 2012 i686  (none))"
ts=2022-03-24T13:47:33.260Z caller=main.go:523 level=info fd_limits="(soft=1024, hard=1024)"
ts=2022-03-24T13:47:33.260Z caller=main.go:524 level=info vm_limits="(soft=2147483647b, hard=2147483647b)"
ts=2022-03-24T13:47:33.265Z caller=web.go:540 level=info component=web msg="Start listening for connections" address=[::1]:9090
ts=2022-03-24T13:47:33.272Z caller=main.go:755 level=error msg="Unable to start web listener" err="listen tcp [::1]:9090: socket: address family not supported by protocol"

**Reply by lujiacn** [reply]
Just facing the same issue, for i386 Linux. Build the 386 package with go1.6 resolve the issue. Go1.7, 1.8 compiled 386 package not work. No issue for x64 arch Linux.

**Reply by roidelapluie** [reply]
Do you mean 1.16, 1.17 and 1.18?

**Reply by lujiacn** [reply]
sorry, 1.16,1.17,1.18
